### PR TITLE
Disambiguate "roll move" shortcuts from "roll oracle" shortcuts

### PR DESCRIPTION
**Pull request by rsek**
this also adds an optional "icon" property to the Tabs element. here, it's used to reinforce the distinction between the "roll a move" and "roll an oracle" icons on the move sheet.

marking this as a draft for now, but that's a "just in case" than anything -- not much in this PR to break, what could possibly go wrong? ;)

### CODE REVIEW

**Comment by rsek** [comment]
Screen Shot 2022-09-15 at 1 49 56 AM
Screen Shot 2022-09-15 at 1 51 00 AM
Screen Shot 2022-09-15 at 1 52 33 AM

**Comment by rsek** [comment]
* in the interest of further distinguishing between oracle rolls and move rolls, i've added some appropriate tooltips on both vue buttons (at the level of the component, when possible), and in the message templates
* added new localizable strings to support the above
* standardized some of the elements/classes in handlebars message templates, too, to make it easier to select/style them consistently in the future
* did a cursory pass on tooltip positioning w/ `data-tooltip-direction`
* `$t` helper method now uses `game.i18n.format()` instead of `game.i18n.localize()` if it's provided with a second argument, and has a little tsdoc annotation to explain this

File: src/module/vue/components/buttons/btn-rollprogress.vue [hunk 1]
```diff
@@ -20,6 +21,16 @@ const props = defineProps<{ item: any; tooltip?: string; disabled?: boolean }>()
...
 
+const progressScore = computed(() => {
+  let item: IronswornItem | undefined
+  if ($actor) {
+    item = $actor.items.find((x) => x.id === props.item._id)
+  } else {
+    item = game.items?.get(props.item.id)
+  }
+  return Math.floor(item?.data.data.current / 4)
+})
```
Comment (ben) [inline src/module/vue/components/buttons/btn-rollprogress.vue:32]: This is always used inside a `<progress-box>` component, which [provides](https://github.com/ben/foundry-ironsworn/blob/34158fa789bf3848cd38f8102253bb9f356f1d76/src/module/vue/components/progress/progress-box.vue#L115) a Foundry item for you:

```suggestion
const $actor = inject($ActorKey, undefined)
const $item = inject($ItemKey, undefined)

const progressScore = computed(() => {
  return Math.floor($item?.data.data.current / 4)
})
```

File: src/module/vue/components/buttons/btn-rollmove.vue [hunk 2]
```diff
@@ -12,25 +12,43 @@
...
-  tooltip?: string
+  move?: Move
   disabled?: boolean
 }>()
 
+const tooltip = computed(() => {
+  let str = game.i18n.format('IRONSWORN.RollMove', {
+    title: props.move?.displayName,
+  })
+  // const { Title, Page } = props.move?.dataforgedMove?.Source ?? {}
```
Comment (ben) [inline src/module/vue/components/buttons/btn-rollmove.vue:31]: Still deciding on which of these to use? It'd be _nice_ to use the page numbers somewhere, but it's not easy to find a good place for them.

File: system/templates/rolls/ironsworn-roll-message.hbs [hunk 3]
```diff
@@ -10,49 +13,60 @@
...
       </span>
     </h4>
 
     {{#if moveOutcome}}
-    <section class='move-outcome'>
-      {{{enrichMarkdown moveOutcome}}}
-    </section>
+      <section class='move-outcome'>
+        {{{enrichMarkdown moveOutcome}}}
+      </section>
```
Comment (ben) [inline system/templates/rolls/ironsworn-roll-message.hbs:28]: OK, I think I've got my VS Code configured to use prettier for handlebars, which seems to match this whitespace preference. Hopefully we won't keep doing this in the future. :D

**Review by ben — COMMENTED** [review]
Wow, this is a wide-ranging PR. For future reference, it'd be easier to review if this was split into 2-4 separate ones.

But I like it. It's good. I only have a couple of notes.

File: src/module/vue/components/buttons/btn-rollmove.vue (hunk 2 above)
Comment (rsek) [reply src/module/vue/components/buttons/btn-rollmove.vue:31]: > Still deciding on which of these to use? It'd be _nice_ to use the page numbers somewhere, but it's not easy to find a good place for them.

yeah. with the introduction of PDF JEPs, there's maybe even a direct use case for them (maybe even the possibility of direct links if the API exposes a way to point to a specific page)... but that's *definitely* a different PR.

for now: maybe some kind of secondary text in the move rules display? like a footer with something like "Source: *Ironsworn Rulebook*, p. XX"

File: system/templates/rolls/ironsworn-roll-message.hbs (hunk 3 above)
Comment (rsek) [reply system/templates/rolls/ironsworn-roll-message.hbs:28]: reckon it'd be do a separate prettier formatting PR to catch all of those in one go?

File: src/module/vue/components/buttons/btn-rollmove.vue (hunk 2 above)
Comment (rsek) [reply src/module/vue/components/buttons/btn-rollmove.vue:31]: on second thought -- that adding a footer to the move text rendering could be its own PR, if you consider losing the `title` page numbering acceptable change in the interim. i'll remove the comment here and add an appropriate TODO elsewhere to note it (and then start the PR to add it).

File: src/module/vue/components/buttons/btn-rollprogress.vue (hunk 1 above)
Comment (rsek) [reply src/module/vue/components/buttons/btn-rollprogress.vue]: oh! yeah, that makes way more sense. yeah, i . given that the same function is coming up twice --

Comment (rsek) [reply src/module/vue/components/buttons/btn-rollprogress.vue]: oh! right, that makes way more sense. given that the same procedure is showing up multiple times -- what do you think of adding an `Item.progressScore()` method to standardize it? or is that not worth it if it's useless for the other item types?

File: src/module/vue/components/buttons/btn-rollmove.vue (hunk 2 above)
Comment (ben) [reply src/module/vue/components/buttons/btn-rollmove.vue:31]: Yeah, that's fine. I think the way the code works now provides the most useful tooltip for where it is.

File: src/module/vue/components/buttons/btn-rollprogress.vue (hunk 1 above)
Comment (ben) [reply src/module/vue/components/buttons/btn-rollprogress.vue]: Yeah, that might be nice. It'd have to return `number | undefined` which is unfortunate, but 🤷🏼

File: src/module/vue/components/buttons/btn-rollmove.vue (hunk 2 above)
Comment (ben) [reply src/module/vue/components/buttons/btn-rollmove.vue:31]: To clarify: maybe delete the commented-out code here, and this should be ready to merge?

**Pull request merged**